src/hugr/views/petgraph.rs

```diff
@@ -6,12 +6,141 @@ use crate::hugr::HugrView;
 use crate::ops::handle::NodeHandle;
 use crate::ops::OpType;
 use crate::types::EdgeKind;
-use crate::{Node, Port};
+use crate::{Hugr, Node, Port};
 
 use context_iterators::{ContextIterator, IntoContextIterator, MapWithCtx};
 use petgraph::visit as pv;
 use portgraph::NodeIndex;
 
+/// A trait grouping the multiple petgraph traits implemented on Hugrs
+//
+// TODO: We could just require these in `HugrView`.
```

Comment:
Ok, how about
```
struct PetgraphWrapper<T: HugrView>(T)

impl<T:HugrView> pv::XYZ for PetgraphWrapper<T> {...} // for each pv::XYZ
```
i.e. you'd need to build a PetgraphWrapper around each HugrView, but we could move most of those impls out of the macro altogether, and use the same struct for a Hugr or `&Hugr`...